aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -203,6 +204,36 @@ extern "C" void strsm_(char *side, char *uplo, char *trans, char *diag, int *n,
 #endif
 
 namespace at {
+namespace meta {
+
+TORCH_META_FUNC(triangular_solve)(const Tensor& self, const Tensor& A, bool upper, bool transpose, bool unitriangular) {
+  TORCH_CHECK(self.dim() >= 2,
+           "torch.triangular_solve: Expected b to have at least 2 dimensions, but it has ", self.dim(), " dimensions instead");
+  TORCH_CHECK(A.dim() >= 2,
+           "torch.triangular_solve: Expected A to have at least 2 dimensions, but it has ", A.dim(), " dimensions instead");
+
+  Tensor self_broadcasted, A_broadcasted;
+  std::tie(self_broadcasted, A_broadcasted) = at::native::_linalg_broadcast_batch_dims(self, A, "triangular_solve");
```

Comment:
This function could benefit from precomputed `Tensor` values (we wouldn't have to call `_linalg_broadcast_batch_dims` two times. But, AFAIU, they aren't yet supported, right @ezyang ?

Reply:
Precomputed values didn't land, but Tensors should never be precomputed!

Reply:
You only make use of sizes() from the broadcasted elements. `_linalg_broadcast_batch_dims` should get refactored to have a variant that only returns sizes.